pkg/interfaces/contracts/vault/IVaultErrors.sol

```diff
@@ -284,6 +285,43 @@ interface IVaultErrors {
      */
     error PoolPauseWindowExpired(address pool);
 
+    /*******************************************************************************
+                            Yield-bearing token buffers
+    *******************************************************************************/
+
+    /// @dev Buffer for the given wrapped token was already initialized.
+    error BufferAlreadyInitialized(IERC4626 wrappedToken);
+
+    /// @dev Buffer for the given wrapped token was not initialized.
+    error BufferNotInitialized(IERC4626 wrappedToken);
+
+    /// @dev The user is trying to remove more than their allocated shares from the buffer.
+    error NotEnoughBufferShares();
+
+    /// @dev The wrapped token asset does not match the underlying token.
+    error WrongWrappedTokenAsset(IERC4626 wrappedToken, address underlyingToken);
+
+    /// @dev A wrapped token reported the zero address as its underlying token asset.
+    error InvalidUnderlyingTokenAsset();
```

Comment:
I guess it's not related to this PR, but underlying token and asset is the same thing. Maybe this error should be `InvalidUnderlyingToken` or `InvalidAsset` (or even `InvalidWrapperAsset`, to reference the function `wrapper.asset()`).

Reply:
InvalidERC4626Asset sounds good, but we use underlyingToken all over the place in the API, so maybe InvalidUnderlyingToken is best to avoid a massive rename

Reply:
yeah I thought about this one. I think the confusion arises from getting `underlyingToken` when calling `wraped.asset()`.
Let's go for `InvalidUnderlyingToken` I guess.